public/app/features/explore/RichHistory/RichHistoryCard.tsx

```diff
@@ -183,8 +183,22 @@ export function RichHistoryCard(props: Props) {
   };
 
   const onDeleteQuery = () => {
-    updateRichHistory(query.ts, 'delete');
-    appEvents.emit(AppEvents.alertSuccess, ['Query deleted']);
+    // For starred queries, we want confirmation. For non-starred, we don't.
+    if (query.starred) {
+      appEvents.emit(CoreEvents.showConfirmModal, {
+        title: 'Delete',
+        text: 'Are you sure you want to permanently delete your starred query?',
```

Comment:
Just a matter of opinion but I've usually seen such things referred to as "favorites" in other applications. Not sure how it is elsewhere in grafana (UI/docs, not code) though.
Which is why I'd suggest changing this to "... delete this favorite query" if it wouldn't be inconsistent. :)

Reply:
Yes, very valid point, however we use starred instead of favourite across the whole grafana - starred queries, starred dashboards.

Reply:
Fair enough, let's leave it then. :)
Otherwise, it looks good to me. :+1: